### DISCUSSION TITLE
Workflow skips step

### DISCUSSION THREAD

**Discussion by HarshalRathore** [community/community#21427] [Other Feature Feedback, Questions, & Ideas]
<p>I am literally trying to debug this thing for a month now. I know it is a small and may be rookie problem but I give up. So here it goes like this, I want to check for broken links on my website and if the workflow finds any I want it to create issue. Note that I want it to create issue only if it founds broken link. If the action that is checking for broken links founds any broken links it returns failure/false. so I wrote this condition <code>if: failure() &amp;&amp; steps.check.outcome == 'failure'</code><br>
So create issue step only executes if the link checking step fails. But skips the link checking step itself I tried to look at the logs it said something like this.<br>
<code>DEBU[0002] Evaluating 'Boolean(failure() &amp;&amp; steps['check']['outcome'] == 'failure')' instead of 'Boolean(failure() &amp;&amp; steps.check.outcome == 'failure')'  DEBU[0002] expression 'failure() &amp;&amp; steps.check.outcome == 'failure'' evaluated to 'false'  DEBU[0002] Skipping step 'victoriadrake/link-snitch@master' due to 'failure() &amp;&amp; steps.check.outcome == 'failure''</code><br>
I tried to debug it by change the condition to <code>if: failure()</code> then also it skipped that step<br>
code.<br>
`name: Link Snitch</p>
<p>on:<br>
schedule:<br>
- cron:  ‘0 1 1 * *’<br>
workflow_dispatch:</p>
<p>env:<br>
URL: <a href="https://open-horizon.github.io/" rel="noopener nofollow ugc">https://open-horizon.github.io/</a><br>
ISSUE_TEMPLATE: “.github/ISSUE_TEMPLATE/check-broken-links.md”</p>
<p>jobs:<br>
build:</p>
<pre><code>runs-on: ubuntu-latest

steps:
  - name: Check out master
    uses: actions/checkout@master
    with:
      fetch-depth: 1
  
  
  - id: check
    uses: victoriadrake/link-snitch@master
    with:
      Resursive urls: true
      URL: https://open-horizon.github.io/

    if: failure() &amp;&amp; steps.check.outcome == 'failure'
  - uses: JasonEtco/create-an-issue@v2.4.0
    env:
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    with:
      filename: ${{ env.ISSUE_TEMPLATE }}`</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="HarshalRathore" data-post="1" data-topic="205370">
<div class="title">
<div class="quote-controls"></div>
 HarshalRathore:</div>
<blockquote>
<pre><code class="lang-yaml">  - id: check
    uses: victoriadrake/link-snitch@master
    with:
      Resursive urls: true
      URL: https://open-horizon.github.io/

    if: failure() &amp;&amp; steps.check.outcome == 'failure'
  - uses: JasonEtco/create-an-issue@v2.4.0
    env:
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    with:
      filename: ${{ env.ISSUE_TEMPLATE }}`
</code></pre>
</blockquote>
</aside>
<p>The problem is that the <code>if</code> is part of the <code>check</code> step itself. The line <em>after</em> the <code>if</code> starts the next item in the YAML list <code>steps</code>. That never matches because at the time when the runner checks if the <code>check</code> step should run, it doesn’t have an outcome yet, and nothing failed unless there’s an error during checkout. :slightly_smiling_face:</p>
<p>If you want the condition to apply to the <code>create-an-issue</code> step, you need to move the condition into that step (note the location of the dash!):</p>
<pre><code class="lang-yaml">  - id: check
    uses: victoriadrake/link-snitch@master
    with:
      Resursive urls: true
      URL: https://open-horizon.github.io/

  - if: failure() &amp;&amp; steps.check.outcome == 'failure'
    uses: JasonEtco/create-an-issue@v2.4.0
    env:
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    with:
      filename: ${{ env.ISSUE_TEMPLATE }}`
</code></pre>
